pkg/cachedinstances/k8s/k8s_test.go

```diff
@@ -241,64 +229,24 @@ func TestIPToTags(t *testing.T) {
 }
 
 func TestNoHostNetworkPodsCached(t *testing.T) {
+	t.Parallel()
+
 	setupTest(t, func(t *testing.T, fixtures *testFixture) {
 		hostNetworkPod := pod()
 		hostNetworkPod.Status.HostIP = ipAddr
 		hostNetworkPod.Spec.HostNetwork = true
 		fixtures.podsWatch.Add(hostNetworkPod)
 		waitForFullCache(t, fixtures, 1)
 
-		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
-		defer cancel()
-
-		instanceData, err := fixtures.cloudProvider.Instance(ctx, ipAddr)
-		require.NoError(t, err)
-
-		instance := instanceData[ipAddr]
+		instance, cacheHit := fixtures.provider.Peek(ipAddr)
+		require.True(t, cacheHit)
 		require.Nil(t, instance)
 	}, viper.New(), nodeName)
 }
 
-func TestWatchNodeOnly(t *testing.T) {
-	v := viper.New()
-	v.Set(ParamWatchCluster, false)
-
-	setupTest(t, func(t *testing.T, fixtures *testFixture) {
-		expectedKey := "node"
-		nodeAnnotationKey := AnnotationPrefix + expectedKey
-		expectedValue := nodeName
-
-		p := pod()
-		p.Spec.NodeName = nodeName
-		p.ObjectMeta.Annotations = map[string]string{nodeAnnotationKey: expectedValue}
-		p.ObjectMeta.Labels = map[string]string{}
-		fixtures.podsWatch.Add(p)
-
-		p = pod2()
-		p.Spec.NodeName = "node2"
-		p.ObjectMeta.Annotations = map[string]string{nodeAnnotationKey: "node2"}
-		p.ObjectMeta.Labels = map[string]string{}
-		fixtures.podsWatch.Add(p)
-
-		waitForFullCache(t, fixtures, 2)
-
-		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
-		defer cancel()
-
-		instanceData, err := fixtures.cloudProvider.Instance(ctx, ipAddr)
```

Comment:
So we ask about `ipAddr` only but then check that the returned map does not have `ipAddr2`. Of course it does not. I.e. bug in test. It always fails because filtering is done server-side.